renderdoc/driver/gl/gl_replay_egl.cpp

```diff
@@ -94,6 +222,159 @@ ReplayCreateStatus GL_CreateReplayDevice(const char *logfile, IReplayDriver **dr
 {
   RDCDEBUG("Creating an OpenGL ES replay device");
 
-  // TODO
-  return eReplayCreate_APIInitFailed;
+  // Query the required EGL functions
+  if(eglCreateContextProc == NULL)
+  {
+    eglGetProcAddressProc = (PFN_eglGetProcAddress)dlsym(RTLD_NEXT, "eglGetProcAddress");
+    eglChooseConfigProc = (PFN_eglChooseConfig)dlsym(RTLD_NEXT, "eglChooseConfig");
+    eglInitializeProc = (PFN_eglInitialize)dlsym(RTLD_NEXT, "eglInitialize");
+    eglBindAPIProc = (PFN_eglBindAPI)dlsym(RTLD_NEXT, "eglBindAPI");
+    eglGetDisplayProc = (PFN_eglGetDisplay)dlsym(RTLD_NEXT, "eglGetDisplay");
+    eglCreateContextProc = (PFN_eglCreateContext)dlsym(RTLD_NEXT, "eglCreateContext");
+    eglMakeCurrentProc = (PFN_eglMakeCurrent)dlsym(RTLD_NEXT, "eglMakeCurrent");
+    eglSwapBuffersProc = (PFN_eglSwapBuffers)dlsym(RTLD_NEXT, "eglSwapBuffers");
+    eglDestroyContextProc = (PFN_eglDestroyContext)dlsym(RTLD_NEXT, "eglDestroyContext");
+    eglDestroySurfaceProc = (PFN_eglDestroySurface)dlsym(RTLD_NEXT, "eglDestroySurface");
+    eglQuerySurfaceProc = (PFN_eglQuerySurface)dlsym(RTLD_NEXT, "eglQuerySurface");
+    eglCreatePbufferSurfaceProc =
+        (PFN_eglCreatePbufferSurface)dlsym(RTLD_NEXT, "eglCreatePbufferSurface");
+    eglCreateWindowSurfaceProc =
+        (PFN_eglCreateWindowSurface)dlsym(RTLD_NEXT, "eglCreateWindowSurface");
+
+    if(eglGetProcAddressProc == NULL || eglBindAPIProc == NULL || eglGetDisplayProc == NULL ||
+       eglCreateContextProc == NULL || eglMakeCurrentProc == NULL || eglSwapBuffersProc == NULL ||
+       eglDestroyContextProc == NULL || eglDestroySurfaceProc == NULL ||
+       eglQuerySurfaceProc == NULL || eglCreatePbufferSurfaceProc == NULL ||
+       eglCreateWindowSurfaceProc == NULL || eglChooseConfigProc == NULL)
+    {
+      RDCERR(
+          "Couldn't find required function addresses, eglGetProcAddress eglCreateContext"
+          "eglSwapBuffers (etc.)");
+      return eReplayCreate_APIInitFailed;
+    }
+  }
+
+  GLInitParams initParams;
+  RDCDriver driverType = RDC_OpenGL;
```

Comment:
Yes, I'm planning adding a new driver type (in fact I'm basically done with it). However I did not wanted to add that "extra code" into this PR.